typing/typecore.ml

```diff
@@ -1287,6 +1299,12 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
           Some (p0, p, true), ty
         with Not_found -> None, newvar ()
       in
+      check_missed_punning_opportunity
+        (function
+          | (lab, {ppat_loc=loc; ppat_desc=Ppat_var x}) ->
+              Some ({lab with txt = Longident.last lab.txt}, loc, x.txt)
```

Comment:
I think the warning only make sense if there are no attributes attached to the pattern/exp